plugins/es-repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureBlobStore.java

```diff
@@ -56,12 +56,8 @@ public AzureBlobStore(RepositoryMetadata metadata) {
     AzureBlobStore(RepositoryMetadata metadata, AzureStorageService service) {
         this.container = Repository.CONTAINER_SETTING.get(metadata.settings());
         this.locationMode = Repository.LOCATION_MODE_SETTING.get(metadata.settings());
-
-        AzureStorageSettings repositorySettings = AzureStorageSettings
-            .getClientSettings(metadata.settings());
-        service.refreshSettings(repositorySettings);
-
         this.service = service;
+        this.service.refreshSettings(service.storageSettings);
```

Comment:
Why is this needed? Afaik this will not refresh any new setting but will refresh with the existing settings.

Reply:
The related `AzureBlobStoreContainerTests` adds the MockSettings to the service. To be able to pick these up the refresh i added this change otherwise the whole `AzureBlobStoreContainerTests` is failing.

https://github.com/crate/crate/blob/77d2cf793f53739290d65be896b134618608a144/plugins/es-repository-azure/src/test/java/org/elasticsearch/repositories/azure/AzureBlobStoreContainerTests.java#L37-L40

Reply:
This change makes the `AzureBlobStoreContainerTests` work.

Reply:
Ok but what is about the `locationMode` setting? With this change it will not be passed to the `AzureStorageService` anymore while it is needed there. See also https://github.com/original-brownbear/elasticsearch/blob/1b9c9058b23e229e165ea265538035deb1fdd259/plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureBlobStore.java#L61

Reply:
btw I also tested locally without the call to `refreshSettings` and the tests pass.

Reply:
There are multiple commits in between on `AzureBlobStore`, so it can't be compared 1-1. From what I can see
`refreshSettings` 
https://github.com/crate/crate/blob/77d2cf793f53739290d65be896b134618608a144/plugins/es-repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureStorageService.java#L134-L136 
calls `AzureStorageSettings.copy(clientSettings);` which copies and sets the location mode.

Reply:
Do you think this is broken and needs additional additional handling/testing?

Reply:
> calls AzureStorageSettings.copy(clientSettings); which copies and sets the location mode.

fine, but the new setting (metadata.settings())  is not passed to it, only the existing settings. So any change for `locationMode` won't be processed afaik, not?

Reply:
The latest commit sets the location from the service settings explicitly. Let me check for some relevant tests.

Reply:
It's a constructor which is only called by tests or `public AzureBlobStore(RepositoryMetadata metadata)` where the settings for metadata and service are the same. I think we are safe here. The additional location handling code is there for the client caching which we removed.

Reply:
👍 Thanks for the clarification.